library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/process/traversal/strategy/optimisation/GafferPopGraphStepStrategy.java

```diff
@@ -24,28 +24,60 @@
 import org.apache.tinkerpop.gremlin.process.traversal.step.map.GraphStep;
 import org.apache.tinkerpop.gremlin.process.traversal.step.map.NoOpBarrierStep;
 import org.apache.tinkerpop.gremlin.process.traversal.strategy.AbstractTraversalStrategy;
+import org.apache.tinkerpop.gremlin.process.traversal.strategy.decoration.OptionsStrategy;
 import org.apache.tinkerpop.gremlin.process.traversal.util.TraversalHelper;
+import org.opencypher.gremlin.translation.CypherAst;
+import org.opencypher.gremlin.translation.translator.Translator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import uk.gov.gchq.gaffer.tinkerpop.process.traversal.step.GafferPopGraphStep;
 
+import java.util.Map;
+import java.util.Optional;
+
 /**
  * The {@link GraphStep} strategy for GafferPop, this will replace the default
  * {@link GraphStep} of a query to add Gaffer optimisations. Such as gathering
  * any {@link HasStep}s so that a Gaffer View can be constructed for the query.
+ * Will also handle the translation any Cypher queries passed in a with() step
+ * into a Gremlin traversal.
  *
  * <pre>
  * g.V().hasLabel()    // replaced by GafferPopGraphStep
  * g.E().hasLabel()    // replaced by GafferPopGraphStep
+ * g.with("cypher", "query") // translated to Gremlin traversal
  * </pre>
  */
 public final class GafferPopGraphStepStrategy extends AbstractTraversalStrategy<TraversalStrategy.ProviderOptimizationStrategy> implements TraversalStrategy.ProviderOptimizationStrategy {
+    private static final Logger LOGGER = LoggerFactory.getLogger(GafferPopGraphStepStrategy.class);
     private static final GafferPopGraphStepStrategy INSTANCE = new GafferPopGraphStepStrategy();
 
+    /**
+     * Key used in a with step to include a opencypher query traversal
+     */
+    public static final String CYPHER_KEY = "cypher";
+
     private GafferPopGraphStepStrategy() {
     }
 
     @Override
     public void apply(final Admin<?, ?> traversal) {
+        // Check for any options on the traversal
+        Optional<OptionsStrategy> optionsStrategy = traversal.getStrategies().getStrategy(OptionsStrategy.class);
+        if (optionsStrategy.isPresent()) {
+            Map<String, Object> options = optionsStrategy.get().getOptions();
+            // Translate and add a cypher traversal in if that key has been set
+            if (options.containsKey(CYPHER_KEY)) {
```

Comment:
It feels like this could be more concise and avoid the nested if.
```suggestion
        // Translate and add a cypher traversal in if that key has been set
        if (optionsStrategy.isPresent() && optionsStrategy.get().getOptions().containsKey(CYPHER_KEY)) {
```

Reply:
This was done like this in case more options were added in future but I think they will probably be handled in the `GafferPopGraphStep` class instead so happy to change.